### PR TITLE
Catch unexpected sql errors

### PR DESCRIPTION
**Pull request by inaod568**
Seems like the validation doesn't catch all weird characters in emails so I catch the exceptions where fitting. This is of course maybe not the best solution in the long run but now, as long as it works, lets use it. 

- [x] Tests passing
- [x] Black formatting
- [ - ] Migrations for any changes to the database schema
- [ ] Rebase/merge the `dev` branch
- [ - ] Note in the CHANGELOG

### CODE REVIEW

File: dds_web/api/user.py [hunk 1]
```diff
@@ -79,9 +80,14 @@ def post(self):
         send_email = json_info.get("send_email", True)
 
         # Check if email is registered to a user
-        existing_user = user_schemas.UserSchema().load({"email": email})
-        unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
+        flask.current_app.logger.debug("Here")
```
Comment (inaod568) [inline dds_web/api/user.py:83]:
```suggestion
```

File: dds_web/api/user.py [hunk 2]
```diff
@@ -79,9 +80,13 @@ def post(self):
...
         # Check if email is registered to a user
-        existing_user = user_schemas.UserSchema().load({"email": email})
-        unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
+        try:
+            existing_user = user_schemas.UserSchema().load({"email": email})
+            unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
+            raise ddserr.DatabaseError(message=str(err), alt_message="Unexpected database error.")
 
+        flask.current_app.logger.debug("Here")
```
Comment (inaod568) [inline dds_web/api/user.py:89]:
```suggestion
```

File: dds_web/api/dds_decorators.py [hunk 3]
```diff
@@ -84,6 +85,8 @@ def make_commit(*args, **kwargs):
         # Run function, catch errors
         try:
             result = func(*args, **kwargs)
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (talavis) [inline dds_web/api/dds_decorators.py:88]: As I learned while writing the code for the file removal, it's actually `sqlalchemy.exc.OperationalError` that is thrown, even though `pymysql.err.OperationalError` is the one shown.

Comment (inaod568) [reply dds_web/api/dds_decorators.py:88]: Oh ok! Thanks, I was wondering about this. I'll change it.

Comment (inaod568) [reply dds_web/api/dds_decorators.py:88]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/api/project.py [hunk 4]
```diff
@@ -609,8 +610,18 @@ def post(self):
             for user in p_info["users_to_add"]:
                 try:
                     existing_user = user_schemas.UserSchema().load(user)
-                except marshmallow.exceptions.ValidationError as err:
-                    addition_status = f"Error for {user.get('email')}: {err}"
+                except (
+                    marshmallow.exceptions.ValidationError,
+                    pymysql.err.OperationalError,
+                    sqlalchemy.exc.SQLAlchemyError,
```
Comment (inaod568) [inline dds_web/api/project.py:616]:
```suggestion
                    marshmallow.exceptions.ValidationError,
                    sqlalchemy.exc.OperationalError,
```

File: dds_web/api/project.py [hunk 5]
```diff
@@ -609,8 +610,18 @@ def post(self):
...
+                ) as err:
+                    if isinstance(err, pymysql.err.OperationalError) or isinstance(
+                        err, sqlalchemy.exc.SQLAlchemyError
+                    ):
```
Comment (inaod568) [inline dds_web/api/project.py:620]:
```suggestion
                    if isinstance(err, sqlalchemy.exc.OperationalError):
```

File: dds_web/api/user.py [hunk 6]
```diff
@@ -79,8 +80,11 @@ def post(self):
         send_email = json_info.get("send_email", True)
 
         # Check if email is registered to a user
-        existing_user = user_schemas.UserSchema().load({"email": email})
-        unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
+        try:
+            existing_user = user_schemas.UserSchema().load({"email": email})
+            unanswered_invite = user_schemas.UnansweredInvite().load({"email": email})
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (inaod568) [inline dds_web/api/user.py:86]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/api/user.py [hunk 7]
```diff
@@ -619,7 +623,11 @@ def post(self):
         if "email" not in json_input:
             raise ddserr.DDSArgumentError(message="User email missing.")
 
-        user = user_schemas.UserSchema().load({"email": json_input.pop("email")})
+        try:
+            user = user_schemas.UserSchema().load({"email": json_input.pop("email")})
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (inaod568) [inline dds_web/api/user.py:628]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/api/user.py [hunk 8]
```diff
@@ -714,7 +722,11 @@ class DeleteUser(flask_restful.Resource):
     @handle_validation_errors
     def delete(self):
 
-        user = user_schemas.UserSchema().load(flask.request.json)
+        try:
+            user = user_schemas.UserSchema().load(flask.request.json)
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (inaod568) [inline dds_web/api/user.py:727]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/api/user.py [hunk 9]
```diff
@@ -790,7 +802,10 @@ def post(self):
             raise ddserr.DDSArgumentError(message="User email missing.")
 
         # Check if email is registered to a user
-        existing_user = user_schemas.UserSchema().load({"email": user_email})
+        try:
+            existing_user = user_schemas.UserSchema().load({"email": user_email})
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (inaod568) [inline dds_web/api/user.py:807]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/web/user.py [hunk 10]
```diff
@@ -539,6 +540,8 @@ def confirm_self_deletion(token):
                 message=f"User deletion request for {user.username} / {user.primary_email.email} failed due to database error: {sqlerr}",
                 alt_message=f"Deletion request for user {user.username} registered with {user.primary_email.email} failed for technical reasons. Please contact the unit for technical support!",
             )
+        except (pymysql.err.OperationalError, sqlalchemy.exc.SQLAlchemyError) as err:
```
Comment (inaod568) [inline dds_web/web/user.py:543]:
```suggestion
        except sqlalchemy.exc.OperationalError as err:
```

File: dds_web/web/user.py [hunk 11]
```diff
@@ -12,6 +12,7 @@
 import flask_login
 import itsdangerous
 import sqlalchemy
+import pymysql
```
Comment (inaod568) [inline dds_web/web/user.py:15]:
```suggestion
```

File: dds_web/api/user.py [hunk 12]
```diff
@@ -18,6 +18,7 @@
 import structlog
 import sqlalchemy
 import http
+import pymysql
```
Comment (inaod568) [inline dds_web/api/user.py:21]:
```suggestion
```

File: dds_web/api/project.py [hunk 13]
```diff
@@ -14,6 +14,7 @@
 import datetime
 import botocore
 import marshmallow
+import pymysql
```
Comment (inaod568) [inline dds_web/api/project.py:17]:
```suggestion
```

File: dds_web/api/dds_decorators.py [hunk 14]
```diff
@@ -14,6 +14,7 @@
 import structlog
 import sqlalchemy
 import marshmallow
+import pymysql
```
Comment (inaod568) [inline dds_web/api/dds_decorators.py:17]:
```suggestion
```

**Review by aanil — APPROVED** [review]
👍

**Review by zishanmirza — APPROVED** [review]

**Pull request merged**